BattleshipFrontend/src/app/board/board.component.ts

```diff
@@ -9,73 +11,285 @@ import { CdkDragDrop, transferArrayItem   } from '@angular/cdk/drag-drop';
+    this.shipList1.splice(index, 1);
+  }
+
+
+  public updateShip(ship: ShipComponent): ShipComponent {
+    ship.left = this.dragEnd.cellX - this.boardElement.nativeElement.getBoundingClientRect().x;
+    ship.top = this.dragEnd.cellY - this.boardElement.nativeElement.getBoundingClientRect().y;
+    ship.col = this.dragEnd.col;
+    ship.row = this.dragEnd.row;
+    ship.occupiedCoords =
+        ship.size === 5 && !ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col+1, y: ship.row}, {x: ship.col+2, y: ship.row}, {x: ship.col+3, y: ship.row}, {x: ship.col+4, y: ship.row}]
+      : ship.size === 5 &&  ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col, y: ship.row+1}, {x: ship.col, y: ship.row+2}, {x: ship.col, y: ship.row+3}, {x: ship.col, y: ship.row+4}]
+      : ship.size === 4 && !ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col+1, y: ship.row}, {x: ship.col+2, y: ship.row}, {x: ship.col+3, y: ship.row}]
+      : ship.size === 4 &&  ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col, y: ship.row+1}, {x: ship.col, y: ship.row+2}, {x: ship.col, y: ship.row+3}]
+      : ship.size === 3 && !ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col+1, y: ship.row}, {x: ship.col+2, y: ship.row}]
+      : ship.size === 3 &&  ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col, y: ship.row+1}, {x: ship.col, y: ship.row+2}]
+      : ship.size === 2 && !ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col+1, y: ship.row}]
+      : ship.size === 2 &&  ship.isVertical ? [{x: ship.col, y: ship.row}, {x: ship.col, y: ship.row+1}]
+      : ship.size === 1 && !ship.isVertical ? [{x: ship.col, y: ship.row}]
+      : []
+    return ship;
+  }
+
+
+  public dragMoved(event: CdkDragMove): void {
+    this.decreaseZIndex(event.source.element);
+  }
   
-  drop(event: CdkDragDrop<Ship[]>) {
-    console.log(event);
 
-    transferArrayItem(
-      event.previousContainer.data,
-      event.container.data,
-      event.previousIndex,
-      event.currentIndex
-    );
+  private decreaseZIndex(element: ElementRef): void {
+    element.nativeElement.style.zIndex = "-1";
+    let elem = element.nativeElement.children[0] as HTMLElement;
+    elem.style.zIndex = "-1";
   }
+  
+  
+  private increaseZIndex(element: ElementRef): void {
+    element.nativeElement.style.zIndex = "100";
+    let elem = element.nativeElement.children[0] as HTMLElement;
+    elem.style.zIndex = "100";
+  }
+
+
+  public getFinalData(): void {
+    const occupiedCoords: Coordinates[][] = [];
+    this.shipList2.forEach(ship => occupiedCoords.push(ship.occupiedCoords));
 
+    this.playerFinalData = {
+      playerId: 0, // TODO: Hardcoding for now, then --> this.players.find(player => player.id === id)
+      ships: occupiedCoords
+    }
+  }
 
 
-  start() {
-    console.log("Start game...");
+  public startGame(): void {
+    this.getFinalData();
+    console.log(this.playerFinalData)
+    // this.router.navigate(["/game"]);
```

Comment:
first you have to do the same thing for all the players, every one do a post call, and if the post call is true, you can pass to the next player

Reply:
I needed the backend response with all the player info for that. But i can make a simple objects array with hard-coded player's id and name and do it :)